src/enzyme_ad/jax/TransformOps/TestTransformOps.cpp

```diff
@@ -0,0 +1,165 @@
+        continue;
+
+      rewriter.setInsertionPointToStart(forOp.getBody());
+      Value iterNum = rewriter.create<arith::SubIOp>(
+          loc, forOp.getInductionVar(), forOp.getLowerBound());
+      iterNum = rewriter.create<arith::DivSIOp>(loc, iterNum, forOp.getStep());
+
+      Value replacementIV = rewriter.create<arith::MulIOp>(loc, iterNum, step);
+      replacementIV = rewriter.create<arith::AddIOp>(loc, replacementIV, init);
+
+      rewriter.replaceAllUsesWith(ba, replacementIV);
+
+      removed.insert(i);
+      steps.insert({i, step});
+      changed = true;
+    }
+
+    if (!changed)
+      return failure();
+
+    SmallVector<Value> newInits;
+    for (unsigned i = 0; i < numIterArgs; i++)
+      if (!removed.contains(i))
+        newInits.push_back(forOp.getInits()[i]);
+
+    rewriter.setInsertionPoint(forOp);
+    auto newForOp = rewriter.create<scf::ForOp>(loc, forOp.getLowerBound(),
+                                                forOp.getUpperBound(),
+                                                forOp.getStep(), newInits);
+    if (!newForOp.getRegion().empty())
+      newForOp.getRegion().front().erase();
+    assert(newForOp.getRegion().empty());
+    rewriter.inlineRegionBefore(forOp.getRegion(), newForOp.getRegion(),
+                                newForOp.getRegion().begin());
+
+    SmallVector<Value> newYields;
+    for (unsigned i = 0; i < numIterArgs; i++)
+      if (!removed.contains(i))
+        newYields.push_back(yield->getOperand(i));
+
+    rewriter.setInsertionPoint(yield);
+    rewriter.replaceOpWithNewOp<scf::YieldOp>(yield, newYields);
+
+    llvm::BitVector toDelete(numIterArgs + 1);
+    for (unsigned i = 0; i < numIterArgs; i++)
+      if (removed.contains(i))
+        toDelete[i + 1] = true;
+    newForOp.getBody()->eraseArguments(toDelete);
+
+    rewriter.setInsertionPoint(newForOp);
+    unsigned curNewRes = 0;
+    for (unsigned i = 0; i < numIterArgs; i++) {
+      auto result = forOp->getResult(i);
+      if (removed.contains(i)) {
+        if (result.use_empty())
+          continue;
+
+        rewriter.setInsertionPointAfter(forOp.getOperation());
+        Value iterNum = rewriter.create<arith::SubIOp>(
+            loc, forOp.getUpperBound(), forOp.getLowerBound());
+        iterNum =
+            rewriter.create<arith::DivSIOp>(loc, iterNum, forOp.getStep());
+
+        Value afterLoop =
+            rewriter.create<arith::MulIOp>(loc, iterNum, steps[i]);
+        afterLoop =
+            rewriter.create<arith::AddIOp>(loc, afterLoop, forOp.getInits()[i]);
+
+        rewriter.replaceAllUsesWith(result, afterLoop);
+      } else {
+        rewriter.replaceAllUsesWith(result, newForOp->getResult(curNewRes++));
+      }
+    }
+
+    forOp->getParentOp()->dump();
+    rewriter.eraseOp(forOp);
+
+    return success();
+  }
+};
+
+} // namespace transform
+} // namespace mlir
+
+#include "src/enzyme_ad/jax/TransformOps/TestTransformPatterns.cpp.inc"
```

Comment:
Test is a weird name here, maybe RaiseTransformOps?

Reply:
Done